### PR TITLE
feat(efs): allow AccessPoint to set client token

### PR DESCRIPTION
**Pull request by perrylson**
### Reason for this change

The [CfnAccessPoint](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_efs.CfnAccessPoint.html) construct supports client token specification. However, the current L2 implementation of [AccessPoint](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_efs.AccessPoint.html) does not support this property. 

The `disable-update-workflow` option was needed when running the updated integration test.

### Description of changes

Added the `clientToken` prop to the existing props of AccessPoint. 
### Description of how you validated changes

Validated with unit and integration testing. 

### Checklist
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by badmintoncryer — CHANGES_REQUESTED** [review]
Thank you for your contribution! I've added some comments.

File: packages/aws-cdk-lib/aws-efs/test/access-point.test.ts [hunk 1]
```diff
@@ -50,6 +50,35 @@ test('support tags for AccessPoint', () => {
...
+    fileSystem,
+    clientToken: '',
+  },
+  )).toThrow(/The length of \'clientToken\' must range from 1 to 64 characters, got: 0 characters/);
+});
+
+test('throw when client token has a length that is greater than 64', () => {
+  expect(() => new AccessPoint(stack, 'MyAccessPoint', {
+    fileSystem,
+    clientToken: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
```
Comment (badmintoncryer) [inline packages/aws-cdk-lib/aws-efs/test/access-point.test.ts:77]:
```suggestion
    clientToken: 'a'.repeat(65),
```

**Review by badmintoncryer — APPROVED** [review]
Thanks! I am proposing just one refactor, but I approve it.

File: packages/aws-cdk-lib/aws-efs/test/access-point.test.ts (hunk 1 above)
Comment (perrylson) [reply packages/aws-cdk-lib/aws-efs/test/access-point.test.ts]: Updated 👍

File: packages/aws-cdk-lib/aws-efs/lib/access-point.ts [hunk 2]
```diff
@@ -201,6 +210,11 @@ export class AccessPoint extends AccessPointBase {
   constructor(scope: Construct, id: string, props: AccessPointProps) {
     super(scope, id);
 
+    const clientToken = props.clientToken;
+    if ((clientToken?.length === 0 || (clientToken && clientToken.length > 64)) && !Token.isUnresolved(clientToken)) {
```
[the remark this answers is not in the document]
Comment (perrylson) [reply packages/aws-cdk-lib/aws-efs/lib/access-point.ts:214]: An empty string is considered falsy. In the above condition, `clientToken` will evaluate to false for empty strings. Hence, no errors will be thrown.  Only a non-empty string will pass the `clientToken` check. 

That's why the current validation checks for a length of 0 to detect empty string values.

[the remark this answers is not in the document]
Comment (GavinZZ) [reply packages/aws-cdk-lib/aws-efs/lib/access-point.ts:214]: Good catch! I missed the empty string case. Thanks for pointing it out.

**Review by GavinZZ — DISMISSED** [review]

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**